include/DeterministicNoiseSimulator.hpp

```diff
@@ -3,124 +3,132 @@
+                    noiseEffects.push_back(dd::phaseFlip);
+                    break;
+                case 'D':
+                    noiseEffects.push_back(dd::depolarization);
+                    break;
+                case 'I':
+                    noiseEffects.push_back(dd::identity);
+                    break;
+                default:
+                    throw std::runtime_error("Unknown noise operation '" + cNoiseEffects + "'\n");
+            }
+        }
 
-    void initializeNoiseProbabilities(double cGateNoiseProbability, double cAmplitudeDampingProb = -1, double cMultiQubitGateFactor = 2) {
-        noiseProb            = cGateNoiseProbability;
-        noiseProbSingleQubit = cGateNoiseProbability;
-        noiseProbMultiQubit  = cGateNoiseProbability * cMultiQubitGateFactor;
+        // initializeNoiseProbabilities
+        noiseProbSingleQubit = cNoiseProbability;
+        noiseProbMultiQubit  = cNoiseProbability * cMultiQubitGateFactor;
 
-        if (cAmplitudeDampingProb < 0) {
+        if (!cAmpDampingProbability) {
             // Default value for amplitude damping prob is double the general error probability
-            ampDampingProb            = cGateNoiseProbability * 2;
-            ampDampingProbSingleQubit = cGateNoiseProbability * 2;
-            ampDampingProbMultiQubit  = cGateNoiseProbability * 2 * cMultiQubitGateFactor;
+            ampDampingProbSingleQubit = cNoiseProbability * 2;
+            ampDampingProbMultiQubit  = cNoiseProbability * 2 * cMultiQubitGateFactor;
         } else {
-            ampDampingProb            = cAmplitudeDampingProb;
-            ampDampingProbSingleQubit = cAmplitudeDampingProb;
-            ampDampingProbMultiQubit  = cAmplitudeDampingProb * cMultiQubitGateFactor;
+            ampDampingProbSingleQubit = cAmpDampingProbability.value();
+            ampDampingProbMultiQubit  = cAmpDampingProbability.value() * cMultiQubitGateFactor;
         }
 
-        if (noiseProb < 0 || ampDampingProb < 0 || ampDampingProbMultiQubit > 1 || noiseProbMultiQubit > 1) {
+        if (noiseProbSingleQubit < 0 || ampDampingProbSingleQubit < 0 || ampDampingProbMultiQubit > 1 || noiseProbMultiQubit > 1) {
             throw std::runtime_error("Error probabilities are faulty!"
                                      "\n single qubit error probability: " +
                                      std::to_string(noiseProbSingleQubit) +
                                      " multi qubit error probability: " + std::to_string(noiseProbMultiQubit) +
                                      "\n single qubit amplitude damping  probability: " + std::to_string(ampDampingProbSingleQubit) +
                                      " multi qubit amplitude damping  probability: " + std::to_string(ampDampingProbMultiQubit));
         }
+        Simulator<DDPackage>::dd->resize(qc->getNqubits());
     }
 
     std::map<std::string, std::size_t> Simulate([[maybe_unused]] unsigned int shots) override {
```

Comment:
why is `shots` marked `[[maybe_unused]]`?

Reply:
Removed the attribute